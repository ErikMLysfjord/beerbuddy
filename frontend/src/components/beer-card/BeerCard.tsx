import Voter from "../voter/Voter";
import styles from "./BeerCard.module.css";

interface BeerCardInterface {
  name: string;
  brewery: string;
  beer_id: number;
  votes: number;
}

const BeerCard = (props: BeerCardInterface) => {
  return (
    <a href={"./project2/beer/" + props.beer_id}>
      <button aria-label={props.name} className={styles.card}>
        <div className={styles.textWrapper}>
          <h1 className={styles.beerName}>{props.name}</h1>
          <p className={styles.breweryName}>{props.brewery}</p>
        </div>
        <Voter votes={props.votes} />
      </button>
    </a>
  );
};
export default BeerCard;
