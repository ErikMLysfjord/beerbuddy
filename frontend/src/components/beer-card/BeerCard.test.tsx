import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import BeerCard from "./BeerCard";
import { axe } from "jest-axe";

describe("BeerCard", () => {
  it("is accessible", async () => {
    const { container } = render(
      <BeerCard
        name={"Example beer"}
        brewery={"Example brewery"}
        beer_id={10}
        votes={10}
        reaction="unreact"
      />
    );
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(
      <BeerCard
        beer_id={2}
        name={"Example beer"}
        brewery={"Example brewery"}
        votes={10}
        reaction="unreact"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders with correct names", () => {
    const { getByText } = render(
      <BeerCard
        beer_id={2}
        name={"Example beer"}
        brewery={"Example brewery"}
        votes={10}
        reaction="unreact"
      />
    );
    expect(getByText("Example beer")).toBeInTheDocument();
    expect(getByText("Example beer")).toBeInstanceOf(HTMLHeadingElement);
    expect(getByText("Example brewery")).toBeInTheDocument();
    expect(getByText("10")).toBeInTheDocument();
  });

  it("renders with correct link", () => {
    /* This test will fail once we set up beercard to work correctly */
    const { getByRole } = render(
      <BeerCard
        beer_id={2}
        name={"Example beer"}
        brewery={"Example brewery"}
        votes={10}
        reaction="unreact"
      />
    );
    expect(getByRole("link")).toHaveAttribute("href", "./project2/beer/2");
  });
});
