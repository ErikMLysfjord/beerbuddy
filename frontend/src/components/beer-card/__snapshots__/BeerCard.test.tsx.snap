// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BeerCard > renders correctly 1`] = `
<div>
  <a
    aria-label="Example beer"
    class="_card_539764"
    href="./project2/beer/2"
  >
    <div
      class="_textWrapper_539764"
    >
      <h1
        class="_beerName_539764"
      >
        Example beer
      </h1>
      <p
        class="_breweryName_539764"
      >
        Example brewery
      </p>
    </div>
    <div
      class="_wrapper_0f0ab2"
    >
      <button
        aria-label="Upvote this beer"
        class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary ant-btn-icon-only"
        style="filter: brightness(0.8);"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="caret-up"
            class="anticon anticon-caret-up"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="caret-up"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
              />
            </svg>
          </span>
        </span>
      </button>
      <h2
        aria-label="Total score"
        class="_count_0f0ab2"
      >
        10
      </h2>
      <button
        aria-label="Downvote this beer"
        class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary ant-btn-icon-only"
        style="filter: brightness(0.8);"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="caret-down"
            class="anticon anticon-caret-down"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="caret-down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
  </a>
</div>
`;
